### PR TITLE
OCLOMRS-1111 OCL should stop and fail any in progress imports at startup

### PR DESCRIPTION
**Pull request by mseaton**
See:  https://openmrs.atlassian.net/browse/OCLOMRS-1111

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java [hunk 1]
```diff
@@ -93,6 +97,25 @@ public void contextRefreshed() {
 		log.info("Open Concept Lab Module refreshed");
 	}
 
+	/**
+	 * If OpenMRS is unexpectedly shutdown while an OCL Import is in progress, the import record in the database
+	 * can be left in an "in progress" state, even though the import is no longer actually running.
+	 * In this situation, we want to ensure these imports are marked as failed at the next startup, so that
+	 * subsequent import attempts can succeed without errors that imports are already in progress
+	 */
+	protected void markInProgressImportsAsFailed() {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java:106]: Could there be a reason why this method is not private?

File: api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java [hunk 2]
```diff
@@ -93,6 +97,25 @@ public void contextRefreshed() {
...
+		ImportService importService = Context.getService(ImportService.class);
+		List<Import> inProgressImports = importService.getInProgressImports();
+		if (!inProgressImports.isEmpty()) {
+			log.warn("Found " + inProgressImports.size() + " in progress imports at startup");
+			for (Import inProgressImport : inProgressImports) {
+				log.warn("Updating Import#" + inProgressImport.getImportId() + " as failed");
+				importService.failImport(inProgressImport, "System restarted during import");
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java:113]: super nit, but this also could happen if the context is refreshed (ie someone manually loads a module via the old UI) while an import is rerunning.  Just wondering if the message should be "Context refreshed and/or system restarted during import."

**Review by mogoodrich — APPROVED** [review]
LGTM, one comment, feel free to ignore, especially if I'm wrong... :)

File: api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java (hunk 1 above)
Comment (mseaton) [reply api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java]: OK, changed it to private @dkayiwa

File: api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java (hunk 2 above)
Comment (mseaton) [reply api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java]: @mogoodrich thanks I updated the message.

**Review by dkayiwa — APPROVED** [review]

**Pull request merged**